plugins/package-managers/maven/src/main/kotlin/utils/MavenSupport.kt

```diff
@@ -302,11 +302,13 @@ class MavenSupport(private val workspaceReader: WorkspaceReader) : Closeable {
         val remoteRepositories = allRepositories.filterNot {
             // Some (Linux) file URIs do not start with "file://" but look like "file:/opt/android-sdk-linux".
             it.url.startsWith("file:/")
-        }.map { repository ->
-            val proxy = repositorySystemSession.proxySelector.getProxy(repository)
-            val authentication = repositorySystemSession.authenticationSelector.getAuthentication(repository)
-            RemoteRepository.Builder(repository).setAuthentication(authentication).setProxy(proxy).build()
-        }.toSet()
+        }.mapTo(mutableSetOf()) { repository ->
+            with(RemoteRepository.Builder(repository)) {
+                repositorySystemSession.proxySelector.getProxy(repository)?.also(::setProxy)
+                repositorySystemSession.authenticationSelector.getAuthentication(repository)?.also(::setAuthentication)
+                build()
+            }
```

Comment:
I just realized that for the builder pattern, using `apply` probably would have been a bit more idiomatic:

```kotlin
RemoteRepository.Builder(repository).apply {
    repositorySystemSession.proxySelector.getProxy(repository)?.also(::setProxy)
    repositorySystemSession.authenticationSelector.getAuthentication(repository)?.also(::setAuthentication)
}.build()
```

I'll leave it to you whether you want to change this.